### PR TITLE
Venue creation/edit restyle

### PR DESCRIPTION
**Pull request by IvanBuljovcic**

### CODE REVIEW

**Comment by thecadams** [comment]
@IvanBuljovcic can you fix the conflicts for this one?

**Review by 0xdevalias — CHANGES_REQUESTED** [review]
Can you please update this for conflicts/etc when you're able, then re-request review.

Thanks!

File: functions/venue.js [hunk 1]
```diff
@@ -112,9 +113,30 @@ const createVenueData = (data, context) => {
     default:
       break;
   }
+
   return venueData;
 };
 
+const createVenueData_v2 = (data, context) => ({
```
Comment (denisdimitrov) [inline functions/venue.js:120]: Is this backwards compatible? If I create a venue with v2, can I edit it on admin v1?

File: functions/venue.js [hunk 2]
```diff
@@ -467,6 +505,67 @@ exports.updateVenue = functions.https.onCall(async (data, context) => {
...
+  if (data.showGrid) {
+    updated.showGrid = data.showGrid;
+    updated.columns = data.columns;
+  }
+
+  // updated = {
+  //   ...updated,
+  //   showGrid: data.showGrid || false,
+  //   columns: data.columns || 1,
+  // }
```
Comment (denisdimitrov) [inline functions/venue.js:564]: This code can be removed if not needed.

File: src/components/atoms/ImageInput/ImageInput.tsx [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+  const imageError =
+    error?.message ||
+    `File size limit is 2mb. You can shrink images at ${GIF_RESIZER_URL}`;
+
+  const initialRender = useRef<boolean>(true);
+
+  useEffect(() => {
+    initialRender.current = false;
+  });
```
Comment (denisdimitrov) [inline src/components/atoms/ImageInput/ImageInput.tsx:45]: This is going to run on every render, you should probably add empty dependency array and make it run just once.

File: src/components/atoms/ImageInput/ImageInput.tsx [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+  return (
+    <>
+      <S.Wrapper
+        small={small}
+        hasError={!!error?.message}
+        backgroundImage={imageUrl}
+        as="label"
+      >
+        <input
+          accept="image/x-png,image/gif,image/jpeg"
```
Comment (denisdimitrov) [inline src/components/atoms/ImageInput/ImageInput.tsx:60]: I'm not 100% sure about this but I think you might have to add `image/png` as well

File: src/components/atoms/SubmitButton/SubmitButton.tsx [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+const SubmitButton: React.FC<SubmitButtonProps> = ({
+  customClass,
+  loading,
+  editing,
+  templateType,
+}) => {
+  if (loading)
+    return (
+      <div className="spinner-border">
+        <span className="sr-only">Loading...</span>
```
Comment (denisdimitrov) [inline src/components/atoms/SubmitButton/SubmitButton.tsx:15]: I saw this code in Button.tsx as well, maybe you can extract it and reuse it.

File: src/pages/Admin/AdminEditComponent_v2.tsx [hunk 6]
```diff
@@ -0,0 +1,545 @@
...
+              <b>Selected Venue:</b> {venue.name}
+            </div>
+            <Link
+              className="btn btn-primary"
+              target="_blank"
+              to={`/admin/venue/edit/${venue.id}`}
+            >
+              Edit Venue
+            </Link>
+            {isCampVenue(venue) && (
```
Comment (denisdimitrov) [inline src/pages/Admin/AdminEditComponent_v2.tsx:224]: This should check if its camp or partymap, something like this:
`
HAS_ROOMS_TEMPLATES.includes(venue.template)
`

File: src/pages/Admin/Admin_v2.tsx [hunk 7]
```diff
@@ -0,0 +1,454 @@
...
+type VenueListProps = {
+  selectedVenueId?: string;
+  roomIndex?: number;
+};
+
+const VenueList: React.FC<VenueListProps> = ({
+  selectedVenueId,
+  roomIndex,
+}) => {
+  const venues = useSelector((state) => state.firestore.ordered.venues);
```
Comment (denisdimitrov) [inline src/pages/Admin/Admin_v2.tsx:63]: You can check if there is a selector for these if not just create one

File: src/pages/Admin/Admin_v2.tsx [hunk 8]
```diff
@@ -0,0 +1,454 @@
...
+      <ul className="page-container-adminsidebar-venueslist">
+        {venues.map((venue, index) => (
+          <li
+            key={index}
+            className={`${selectedVenueId === venue.id ? "selected" : ""} ${
+              canHaveSubvenues(venue) ? "camp" : ""
+            }`}
+          >
+            <Link to={`/admin/venue/${venue.id}`}>{venue.name}</Link>
+            {isCampVenue(venue) && (
```
Comment (denisdimitrov) [inline src/pages/Admin/Admin_v2.tsx:84]: `HAS_ROOMS_TEMPLATES.includes(venue.template)`

File: src/pages/Admin/Admin_v2.tsx [hunk 9]
```diff
@@ -0,0 +1,454 @@
...
+              <ul className="page-container-adminsidebar-subvenueslist">
+                {venue.rooms.map((room, idx) => (
```
Comment (denisdimitrov) [inline src/pages/Admin/Admin_v2.tsx:86]: I think the typings for the rooms are messed up and this needs to be checked with `venue?.rooms`

File: src/pages/Admin/Venue/VenueWizard/VenueWizard.tsx [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+import VenueWizardCreate from "./Create";
+
+// Hooks
+import { useParams } from "react-router-dom";
+
+// Reducer
+import { VenueWizardReducer, initialState } from "./redux";
+
+const VenueWizard: React.FC = () => {
+  const { venueId } = useParams<{ venueId?: string | undefined }>();
```
Comment (denisdimitrov) [inline src/pages/Admin/Venue/VenueWizard/VenueWizard.tsx:14]: I believe useParams is deprecated/outdated and now we should use `useVenueId()`

File: src/pages/Admin/VenueList/VenueList.tsx [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+      <ul className="page-container-adminsidebar-venueslist">
+        {venues.map((venue, index) => (
+          <li
+            key={index}
+            className={`${selectedVenueId === venue.id ? "selected" : ""} ${
+              canHaveSubvenues(venue) ? "camp" : ""
+            }`}
+          >
+            <Link to={`/admin/venue/${venue.id}`}>{venue.name}</Link>
+            {isCampVenue(venue) && (
```
Comment (denisdimitrov) [inline src/pages/Admin/VenueList/VenueList.tsx:37]: `HAS_ROOMS_TEMPLATES.includes(venue.template)`

File: src/pages/Admin/Admin_v2.tsx [hunk 12]
```diff
@@ -0,0 +1,454 @@
...
+            </Link>
+            {canHaveSubvenues(venue) && (
+              <Link
+                to={`/admin/venue/rooms/${venue.id}`}
+                className="btn btn-block"
+              >
+                Add a Room
+              </Link>
+            )}
+            {isCampVenue(venue) && typeof roomIndex !== "undefined" && (
```
Comment (denisdimitrov) [inline src/pages/Admin/Admin_v2.tsx:340]: `HAS_ROOMS_TEMPLATES.includes(venue.template)`

**Review by denisdimitrov — CHANGES_REQUESTED** [review]

File: functions/venue.js (hunk 1 above)
Comment (IvanBuljovcic) [reply functions/venue.js:120]: Not entirely.
`createVenueData` has more fields that it sends on creation, that `createVenueData_2` does.

For example, `createVenueData` sends `code_of_conduct_questions`

That data is currently not accessible nor editable.

Any data that is shared between the two functions, such as `name`, `showGrid`, `coverImageUrl` can be edited in both versions of the Admin.

File: src/components/atoms/ImageInput/ImageInput.tsx (hunk 4 above)
Comment (IvanBuljovcic) [reply src/components/atoms/ImageInput/ImageInput.tsx:60]: The accept argument was done in a recent PR, ill need to find what was added and copy that over

File: src/pages/Admin/Details/Form/DetailsForm.tsx [hunk 13]
```diff
@@ -0,0 +1,305 @@
...
+  const disable = isSubmitting;
+  const templateID = VenueTemplate.partymap;
+  const nameDisabled = isSubmitting || !!venueId;
+
+  const defaultVenue = createJazzbar({});
+
+  useEffect(() => {
+    if (editData && venueId) {
+      setValue([
+        { name: editData?.name },
```
Comment (mike-lvov) [inline src/pages/Admin/Details/Form/DetailsForm.tsx:105]: I think `editData.name` will work as you have checked for `editData` above

File: src/pages/Admin/Details/Form/DetailsForm.tsx [hunk 14]
```diff
@@ -0,0 +1,305 @@
...
+
+  const values = watch();
+  const { isSubmitting } = formState;
+
+  const urlSafeName = values.name
+    ? `${window.location.host}${venueLandingUrl(
+        createUrlSafeName(values.name)
+      )}`
+    : undefined;
+  const disable = isSubmitting;
```
Comment (mike-lvov) [inline src/pages/Admin/Details/Form/DetailsForm.tsx:96]: bool naming

File: src/pages/Admin/Details/Form/DetailsForm.tsx [hunk 15]
```diff
@@ -0,0 +1,305 @@
...
+  const templateID = VenueTemplate.partymap;
+  const nameDisabled = isSubmitting || !!venueId;
```
Comment (mike-lvov) [inline src/pages/Admin/Details/Form/DetailsForm.tsx:98]: bool naming

File: src/pages/Admin/Details/Form/DetailsForm.tsx [hunk 16]
```diff
@@ -0,0 +1,305 @@
...
+  const values = watch();
+  const { isSubmitting } = formState;
+
+  const urlSafeName = values.name
+    ? `${window.location.host}${venueLandingUrl(
+        createUrlSafeName(values.name)
+      )}`
+    : undefined;
+  const disable = isSubmitting;
+  const templateID = VenueTemplate.partymap;
```
Comment (mike-lvov) [inline src/pages/Admin/Details/Form/DetailsForm.tsx:97]: I think according to naming convention `templateId` should be better

File: src/pages/Admin/Details/Form/DetailsForm.tsx [hunk 17]
```diff
@@ -0,0 +1,305 @@
...
+      payload: {
+        name: values.name,
+        subtitle: values.subtitle,
+        description: values.description,
+      },
+    });
+  };
+
+  return (
+    <S.Form onSubmit={handleSubmit(onSubmit)} onChange={handleOnChange}>
```
Comment (mike-lvov) [inline src/pages/Admin/Details/Form/DetailsForm.tsx:251]: consider wrapping it in useCallback

File: src/pages/Admin/Details/ValidationSchema.ts [hunk 18]
```diff
@@ -0,0 +1,178 @@
...
+import "firebase/functions";
+import {
+  PLAYA_VENUE_SIZE,
+  MAX_IMAGE_FILE_SIZE_BYTES,
+  GIF_RESIZER_URL,
+  PLAYA_WIDTH,
+  PLAYA_HEIGHT,
+} from "settings";
+
+const initialMapIconPlacement: VenueInput["placement"] = {
```
Comment (mike-lvov) [inline src/pages/Admin/Details/ValidationSchema.ts:14]: The naming conventions for consts usually states that they should have their names in form of `INITIAL_MAP_ICON_PLACEMENT`

File: src/pages/Admin/Venue/VenueWizard/Create/VenueWizardCreate.tsx [hunk 19]
```diff
@@ -0,0 +1,49 @@
...
+import { AuthOptions } from "components/organisms/AuthenticationModal/AuthenticationModal";
+
+const VenueWizardCreate: React.FC<VenueWizardCreateProps> = ({
+  state,
+  dispatch,
+}) => {
+  const history = useHistory();
+  const { user } = useUser();
+
+  const previous = useCallback(
```
Comment (mike-lvov) [inline src/pages/Admin/Venue/VenueWizard/Create/VenueWizardCreate.tsx:25]: It is not clear that it is a function from the naming

File: src/pages/Admin/Venue/VenueWizard/redux/reducer.ts [hunk 20]
```diff
@@ -0,0 +1,51 @@
...
+  SetBannerUrl,
+  SetSquareLogoUrl,
+  WizardAction,
+  WizardState,
+  SetFormValues,
+} from "./types";
+
+import * as ActionTypes from "./actionTypes";
+
+export const initialState = {
```
Comment (mike-lvov) [inline src/pages/Admin/Venue/VenueWizard/redux/reducer.ts:11]: export const initialState: WizardState

File: src/pages/Admin/VenueList/VenueList.tsx [hunk 21]
```diff
@@ -0,0 +1,60 @@
...
+// Typings
+import { VenueListProps } from "./VenueList.types";
+import { canHaveSubvenues } from "utils/venue";
+import { HAS_ROOMS_TEMPLATES } from "settings";
+
+const VenueList: React.FC<VenueListProps> = ({
+  selectedVenueId,
+  roomIndex,
+}) => {
+  const venues = useSelector((state) => state.firestore.ordered.venues);
```
Comment (mike-lvov) [inline src/pages/Admin/VenueList/VenueList.tsx:16]: Would you please use a selector here?

**Review by mike-lvov — CHANGES_REQUESTED** [review]

**Pull request closed without merging**